tests/deprecated_api/test_remove_1-5.py

```diff
@@ -123,6 +123,18 @@ def test_v1_5_0_model_checkpoint_period(tmpdir):
         ModelCheckpoint(dirpath=tmpdir, period=1)
 
 
+def test_old_transfer_batch_to_device_hook(tmpdir):
+
+    class OldModel(BoringModel):
+
+        def transfer_batch_to_device(self, batch, device):
+            return super().transfer_batch_to_device(batch, device, None)
```

Comment:
but this model only contains one dataloader? Why do we expect a warning?
`validation_step(batch, batch_idx)` doesn't warn about missing dataloader_idx if I have only one dataloader, right?

Reply:
the arguments are consistent here. In case there is only one dataloader then `dataloader_idx is None`. Although there is some inconsistency with this argument in other methods. For eg in `on_train./val/test_batch_start` this is a required argument but in `validation_step` as you said it's not. So I though making it a required argument makes more sense as it makes method inputs same in all the cases so doesn't need to change their API if number of dataloaders are updated.